### PR TITLE
General: update WP version requirements to WordPress 6.3.

### PR DESCRIPTION
**Pull request by jeherve**
## Proposed changes:

WordPress 6.4 is out, we can now require WP-1, i.e. WP 6.3.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

Primary issue: #32865

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

* Is CI happy?

### CODE REVIEW

File: .phpcs.config.xml [hunk 1]
```diff
@@ -2,8 +2,8 @@
 <!-- This file configures everything except the list of rules. -->
 <!-- The separation is so .github/files/phpcompatibility-dev-phpcs.xml can use the same config with a different rule set. -->
 <ruleset>
-	<config name="minimum_supported_wp_version" value="6.2" />
-	<config name="testVersion" value="5.6-"/>
+	<config name="minimum_supported_wp_version" value="6.3" />
+	<config name="testVersion" value="7.0-"/>
```
Comment (anomiex) [inline .phpcs.config.xml:6]: :tada: 

I'm guessing a separate PR will be for dropping obsolete code paths, but if we're going to do this here then let's go ahead and drop PHP 5 from other places in CI too:

* https://github.com/Automattic/jetpack/blob/133f8dbf11f8506a7bf1f9274b44e98fe738357c/.github/files/generate-ci-matrix.php#L71-L78
* https://github.com/Automattic/jetpack/blob/133f8dbf11f8506a7bf1f9274b44e98fe738357c/.github/versions.sh#L8
* https://github.com/Automattic/jetpack/blob/133f8dbf11f8506a7bf1f9274b44e98fe738357c/.github/workflows/linting.yml#L174

Bunch of doc references to 5.6 too:

* https://github.com/Automattic/jetpack/blob/133f8dbf11f8506a7bf1f9274b44e98fe738357c/docs/coding-guidelines.md?plain=1#L7
* https://github.com/Automattic/jetpack/blob/133f8dbf11f8506a7bf1f9274b44e98fe738357c/docs/development-environment.md?plain=1#L378-L384
* https://github.com/Automattic/jetpack/blob/133f8dbf11f8506a7bf1f9274b44e98fe738357c/docs/monorepo.md?plain=1#L45
* https://github.com/Automattic/jetpack/blob/133f8dbf11f8506a7bf1f9274b44e98fe738357c/docs/monorepo.md?plain=1#L213
* https://github.com/Automattic/jetpack/blob/133f8dbf11f8506a7bf1f9274b44e98fe738357c/docs/monorepo.md?plain=1#L221
* https://github.com/Automattic/jetpack/blob/133f8dbf11f8506a7bf1f9274b44e98fe738357c/docs/testing/regression-checklist/regression-checklist.md?plain=1#L24

File: projects/plugins/backup/readme.txt [hunk 2]
```diff
@@ -1,7 +1,7 @@
 === Jetpack VaultPress Backup ===
 Contributors: automattic, bjorsch, fgiannar, initsogar, jeherve, jwebbdev, kraftbj, macbre, samiff, sermitr, williamvianas
 Tags: jetpack, backup, restore
-Requires at least: 6.2
+Requires at least: 6.3
 Requires PHP: 5.6
```
Comment (anomiex) [inline projects/plugins/backup/readme.txt:5]: Should we update all these "Requires PHP" lines right away too?

File: .phpcs.config.xml (hunk 1 above)
Comment (jeherve) [reply .phpcs.config.xml:6]: > I'm guessing a separate PR will be for dropping obsolete code paths,

Yup, doing this in #34126

File: projects/plugins/backup/readme.txt (hunk 2 above)
Comment (jeherve) [reply projects/plugins/backup/readme.txt:5]: I was hoping #34126 would get merged before this one, that should help.

Comment (anomiex) [reply projects/plugins/backup/readme.txt:5]: 👍

**Review by kraftbj — APPROVED** [review]
Let's do it!

**Pull request merged**